File: seed_recipes.mjs
// Seed built-in recipes to Turso database
// Run from project root: node seed_recipes.mjs

import { createClient } from '@libsql/client';
import { readFileSync } from 'fs';

const TURSO_URL = "libsql://daily-food-chain-tlgeorge.aws-us-east-1.turso.io";
const TURSO_TOKEN = "eyJhbGciOiJFZERTQSIsInR5cCI6IkpXVCJ9.eyJpYXQiOjE3NzI2MjQxOTAsImlkIjoiMDE5Y2I4YTItNTQwMS03Y2VmLWFkMjYtNjAwZWJiYTUwNzkyIiwicmlkIjoiY2ZiMjkyYmYtNjViNC00NTQ1LThiYzgtMjM5ZDVhODQzYzYwIn0.C2ZB6niPeWPmKA16_IzNbSLn8VAOELLjrdml6ZXp12cfKfMtS6f4w1w32LFlUPjiJLiYhg9yGbjzojvUGPqNDg";

// Parse levels from TypeScript file
function parseLevelsFromFile(filepath) {
  const content = readFileSync(filepath, 'utf-8');
  
  // Find LEVELS array (everything between export const LEVELS and the closing ];)
  const levelsMatch = content.match(/export const LEVELS: Level\[\] = \[([\s\S]*?)\n\];/);
  if (!levelsMatch) {
    throw new Error('Could not find LEVELS array in file');
  }
  
  const levelsContent = levelsMatch[1];
  const levels = [];
  
  // Split by level start pattern (line starting with "  {" after a comma or start)
  const levelBlocks = levelsContent.split(/\n  \{/).filter(b => b.includes("id:"));
  
  for (const block of levelBlocks) {
    const level = {};
    
    // Extract string fields
    const stringFields = ['id', 'name', 'category', 'dietaryCategory', 'servings', 'prepTime'];
    for (const field of stringFields) {
      const match = block.match(new RegExp(`${field}: '([^']*)'`));
      if (match) level[field] = match[1];
    }
    
    // Extract levelNum
    const numMatch = block.match(/levelNum: (\d+)/);
    if (numMatch) level.levelNum = parseInt(numMatch[1]);
    
    // Extract recipe array
    const recipeMatch = block.match(/recipe: \[([^\]]*)\]/);
    if (recipeMatch) {
      level.recipe = [...recipeMatch[1].matchAll(/'([^']*)'/g)].map(m => m[1]);
    }
    
    // Extract foodSupply
    const supplyMatch = block.match(/foodSupply: \{([^}]+)\}/);
    if (supplyMatch) {
      level.foodSupply = {};
      for (const [, key, val] of supplyMatch[1].matchAll(/(\w+): (\d+)/g)) {
        level.foodSupply[key] = parseInt(val);
      }
    }
    
    // Extract tools
    level.tools = [...block.matchAll(/\{ type: '(\w+)', count: (\d+), emoji: '([^']+)' \}/g)]
      .map(m => ({ type: m[1], count: parseInt(m[2]), emoji: m[3] }));
    
    // Extract animalSpawns
    level.animalSpawns = [...block.matchAll(/\{ type: '(\w+)', delay: (\d+)/g)]
      .map(m => ({ type: m[1], delay: parseInt(m[2]) }));
    
    // Extract recipeInstructions
    const instrMatch = block.match(/recipeInstructions: \[([\s\S]*?)\]/);
    if (instrMatch) {
      level.recipeInstructions = [...instrMatch[1].matchAll(/'([^']*)'/g)].map(m => m[1]);
    }
    
    // Extract recipeIngredients
    level.recipeIngredients = [...block.matchAll(/\{ name: '([^']+)', quantity: '([^']+)' \}/g)]
      .map(m => ({ name: m[1], quantity: m[2] }));
    
    if (level.id) {
      levels.push(level);
    }
  }
  
  return levels;
}

async function seedToTurso(levels) {
  const client = createClient({
    url: TURSO_URL,
    authToken: TURSO_TOKEN
  });
  
  // Clear existing builtin recipes
  await client.execute("DELETE FROM recipes WHERE type = 'builtin'");
  console.log('🗑️  Cleared existing built-in recipes');
  
  // Insert each level
  for (const level of levels) {
    await client.execute({
      sql: `INSERT INTO recipes (
        id, type, name, category, dietary_category, level_num,
        prep_time, servings,
        recipe, recipe_ingredients, recipe_instructions,
        food_supply, tools, animal_spawns,
        submitted_by, status, created_at
      ) VALUES (?, 'builtin', ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, 'System', 'approved', datetime('now'))`,
      args: [
        `builtin-${level.id}`,
        level.name || '',
        level.category || '',
        level.dietaryCategory || null,
        level.levelNum || null,
        level.prepTime || null,
        level.servings || null,
        JSON.stringify(level.recipe || []),
        JSON.stringify(level.recipeIngredients || []),
        JSON.stringify(level.recipeInstructions || []),
        JSON.stringify(level.foodSupply || {}),
        JSON.stringify(level.tools || []),
        JSON.stringify(level.animalSpawns || [])
      ]
    });
    console.log(`   ✅ ${level.name}`);
  }
  
  // Verify count
  const result = await client.execute("SELECT COUNT(*) as count FROM recipes WHERE type = 'builtin'");
  console.log(`\n🎉 Seeded ${result.rows[0].count} built-in recipes to Turso`);
}

// Main
const tsFile = 'src/lib/farmers-basket/game-state.svelte.ts';
console.log(`📖 Parsing levels from ${tsFile}...`);
const levels = parseLevelsFromFile(tsFile);
console.log(`   Found ${levels.length} levels\n`);

console.log('🌱 Seeding to Turso...');
await seedToTurso(levels);

File: _archived/scripts/seed_recipes.py
#!/usr/bin/env python3
"""
Seed built-in recipes from game-state.svelte.ts to Turso database.
Parses the TypeScript file and extracts all LEVELS entries.
"""

import re
import json
import libsql_experimental as libsql
import os

# Turso connection details
TURSO_URL = "libsql://daily-food-chain-tlgeorge.aws-us-east-1.turso.io"
TURSO_TOKEN = "eyJhbGciOiJFZERTQSIsInR5cCI6IkpXVCJ9.eyJpYXQiOjE3NzI2MjQxOTAsImlkIjoiMDE5Y2I4YTItNTQwMS03Y2VmLWFkMjYtNjAwZWJiYTUwNzkyIiwicmlkIjoiY2ZiMjkyYmYtNjViNC00NTQ1LThiYzgtMjM5ZDVhODQzYzYwIn0.C2ZB6niPeWPmKA16_IzNbSLn8VAOELLjrdml6ZXp12cfKfMtS6f4w1w32LFlUPjiJLiYhg9yGbjzojvUGPqNDg"

def parse_levels_from_ts(filepath: str) -> list[dict]:
    """Extract LEVELS array from TypeScript file."""
    with open(filepath, 'r') as f:
        content = f.read()
    
    # Find LEVELS array
    match = re.search(r'export const LEVELS: Level\[\] = \[(.*?)\];', content, re.DOTALL)
    if not match:
        raise ValueError("Could not find LEVELS array")
    
    levels_str = match.group(1)
    
    # Parse each level object
    levels = []
    # Find each level block starting with {
    level_pattern = re.compile(r'\{[^{}]*(?:\{[^{}]*\}[^{}]*)*\}', re.DOTALL)
    
    # Actually, let's use a simpler approach - extract key fields directly
    level_blocks = re.split(r'\n  \{', levels_str)
    
    for block in level_blocks:
        if "id:" not in block:
            continue
        
        level = {}
        
        # Extract simple string fields
        for field in ['id', 'name', 'category', 'dietaryCategory', 'servings', 'prepTime']:
            match = re.search(rf"{field}: '([^']*)'", block)
            if match:
                level[field] = match.group(1)
        
        # Extract levelNum
        match = re.search(r'levelNum: (\d+)', block)
        if match:
            level['levelNum'] = int(match.group(1))
        
        # Extract recipe array
        match = re.search(r"recipe: \[([^\]]*)\]", block)
        if match:
            recipe_str = match.group(1)
            level['recipe'] = re.findall(r"'([^']*)'", recipe_str)
        
        # Extract foodSupply
        match = re.search(r"foodSupply: \{([^}]+)\}", block)
        if match:
            supply_str = match.group(1)
            supply = {}
            for item in re.findall(r"(\w+): (\d+)", supply_str):
                supply[item[0]] = int(item[1])
            level['foodSupply'] = supply
        
        # Extract tools
        tools = []
        for tool_match in re.finditer(r"\{ type: '(\w+)', count: (\d+), emoji: '([^']+)' \}", block):
            tools.append({
                'type': tool_match.group(1),
                'count': int(tool_match.group(2)),
                'emoji': tool_match.group(3)
            })
        if tools:
            level['tools'] = tools
        
        # Extract animalSpawns
        spawns = []
        for spawn_match in re.finditer(r"\{ type: '(\w+)', delay: (\d+)", block):
            spawns.append({
                'type': spawn_match.group(1),
                'delay': int(spawn_match.group(2))
            })
        if spawns:
            level['animalSpawns'] = spawns
        
        # Extract recipeInstructions
        match = re.search(r"recipeInstructions: \[(.*?)\](?:,|\n)", block, re.DOTALL)
        if match:
            instr_str = match.group(1)
            level['recipeInstructions'] = re.findall(r"'([^']*)'", instr_str)
        
        # Extract recipeIngredients
        ingredients = []
        for ing_match in re.finditer(r"\{ name: '([^']+)', quantity: '([^']+)' \}", block):
            ingredients.append({
                'name': ing_match.group(1),
                'quantity': ing_match.group(2)
            })
        if ingredients:
            level['recipeIngredients'] = ingredients
        
        if 'id' in level:
            levels.append(level)
    
    return levels


def seed_to_turso(levels: list[dict]):
    """Insert levels into Turso database."""
    conn = libsql.connect("daily-food-chain", sync_url=TURSO_URL, auth_token=TURSO_TOKEN)
    conn.sync()
    
    cursor = conn.cursor()
    
    # Clear existing builtin recipes
    cursor.execute("DELETE FROM recipes WHERE type = 'builtin'")
    
    for level in levels:
        cursor.execute("""
            INSERT INTO recipes (
                id, type, name, category, dietary_category, level_num,
                prep_time, servings,
                recipe, recipe_ingredients, recipe_instructions,
                food_supply, tools, animal_spawns,
                submitted_by, status, created_at
            ) VALUES (?, 'builtin', ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, 'System', 'approved', datetime('now'))
        """, [
            f"builtin-{level.get('id', '')}",
            level.get('name', ''),
            level.get('category', ''),
            level.get('dietaryCategory'),
            level.get('levelNum'),
            level.get('prepTime'),
            level.get('servings'),
            json.dumps(level.get('recipe', [])),
            json.dumps(level.get('recipeIngredients', [])),
            json.dumps(level.get('recipeInstructions', [])),
            json.dumps(level.get('foodSupply', {})),
            json.dumps(level.get('tools', [])),
            json.dumps(level.get('animalSpawns', [])),
        ])
    
    conn.commit()
    conn.sync()
    
    # Verify
    cursor.execute("SELECT COUNT(*) FROM recipes WHERE type = 'builtin'")
    count = cursor.fetchone()[0]
    print(f"✅ Seeded {count} built-in recipes to Turso")
    
    return count


if __name__ == "__main__":
    ts_file = "src/lib/farmers-basket/game-state.svelte.ts"
    
    print(f"📖 Parsing levels from {ts_file}...")
    levels = parse_levels_from_ts(ts_file)
    print(f"   Found {len(levels)} levels")
    
    # Debug: print first level
    if levels:
        print(f"   First level: {levels[0]['name']}")
    
    print(f"\n🌱 Seeding to Turso...")
    seed_to_turso(levels)
